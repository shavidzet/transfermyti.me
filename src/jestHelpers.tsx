/* eslint-disable import/no-extraneous-dependencies,@typescript-eslint/no-explicit-any,no-console */
import React from "react";
import {
  RenderOptions,
  RenderResult,
  render as testRender,
} from "@testing-library/react";
import { ThemeProvider } from "emotion-theming";
import { theme } from "./components/emotion";

// These are exported for convenience (so we don't have to add an additional import
// to the test file).
export { createEvent, fireEvent, RenderResult } from "@testing-library/react";
export { theme } from "./components/emotion";

/**
 * Renders the specified component with react-testing-library wrapped in a theme
 * provider so we don't have to do it explicitly for each component.
 */
export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult {
  return testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options);
}

type LogMethod = "log" | "warn" | "error";

export class ConsoleToggle {
  private consoleMethod: (...args: any[]) => void = console.warn;
  private readonly logMethod: LogMethod;

  public constructor(logMethod: LogMethod) {
    this.logMethod = logMethod;
  }

  public onBefore(): void {
    this.consoleMethod = console[this.logMethod];
    console[this.logMethod] = jest.fn();
  }

  public onAfter(): void {
    console[this.logMethod] = this.consoleMethod;
  }
}
