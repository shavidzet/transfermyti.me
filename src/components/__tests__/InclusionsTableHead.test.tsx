import React from "react";
import { render } from "~/jestHelpers";
import InclusionsTableHead from "../InclusionsTableHead";

describe("the <InclusionsTableHead> component", () => {
  test("renders successfully with valid props", () => {
    const wrapper = render(
      <table>
        <InclusionsTableHead labels={["Column A", "Column B"]} />
      </table>,
    );

    expect(wrapper.container.firstElementChild).toMatchSnapshot();
  });
});
