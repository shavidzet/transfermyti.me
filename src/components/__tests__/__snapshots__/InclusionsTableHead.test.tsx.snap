// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <InclusionsTableHead> component renders successfully with valid props 1`] = `
<table>
  <thead>
    <tr>
      <th
        scope="col"
      >
        Column A
      </th>
      <th
        scope="col"
      >
        Column B
      </th>
      <th
        data-include="true"
        scope="col"
      >
        Include?
      </th>
    </tr>
  </thead>
</table>
`;
